src/hera/_yaml.py

```diff
@@ -0,0 +1,31 @@
+from types import ModuleType
+from typing import Optional
+
+_yaml: Optional[ModuleType] = None
+try:
+    import yaml
+
+    _yaml = yaml
+except ImportError:
+    _yaml = None
+else:
+
+    def str_presenter(dumper, data):
+        """Configures yaml for dumping multiline strings.
+
+        Ref: https://github.com/yaml/pyyaml/issues/240
+        """
+        if data.count("\n") > 0:  # check for multiline string
+            return dumper.represent_scalar("tag:yaml.org,2002:str", data, style="|")
+        return dumper.represent_scalar("tag:yaml.org,2002:str", data)
+
+    _yaml.add_representer(str, str_presenter)
+    _yaml.representer.SafeRepresenter.add_representer(str, str_presenter)
+
+
+def dump(*args, default_flow_style: bool = False, sort_keys: bool = False, **kwargs) -> str:
+    """Builds the Workflow as an Argo schema Workflow object and returns it as yaml string."""
+    if not _yaml:
+        raise ImportError("`PyYAML` is not installed. Install `hera[yaml]` to bring in the extra dependency")
+
+    return _yaml.dump(*args, default_flow_style=default_flow_style, sort_keys=sort_keys, **kwargs)
```

Comment:
I'm not sure it's worth having `default_flow_style: bool = False, sort_keys: bool = False` in the function signature as we're passing args/kwargs through to the actual function. We can do something with decorators/`ParamSpec` to help intellisense to get the args/kwargs of the real `dump` function but that can be done later (and might have an issue if you have a named arg that would be expected in the kwargs, so would prefer to remove them for now).

```suggestion
def dump(*args, **kwargs) -> str:
    """Builds the Workflow as an Argo schema Workflow object and returns it as yaml string."""
    if not _yaml:
        raise ImportError("`PyYAML` is not installed. Install `hera[yaml]` to bring in the extra dependency")

    # Set some default options if not provided by the user
    kwargs.setdefault("default_flow_style", False)
    kwargs.setdefault("sort_keys", False)
    return _yaml.dump(*args, **kwargs)
```

Reply:
I dont think the native yaml dump function in either of the yaml libraries have real kwargs typings, just **kwargs, fwiw.

I'm pretty sure there's no functional difference between the two versions in any way (except that you get slightly more signature information). As for it being a named arg or not, if the underlying library changed params, given that the **kwargs are getting passed through from your public API, it'd represent a user visible behavior change. In that way, you're probably better off having explicit options here rather than even passing through the library's own kwargs.

With that said, it's an internal module now, and these named kwargs are not getting passed through from the root `to_yaml` call, so it's mostly irrelevant. I dont feel super strongly about putting back to how it was.